hyperspy/io_plugins/blockfile.py

```diff
@@ -0,0 +1,241 @@
+        raise ValueError('Blockfiles require signal shape to be square!')
+    DP_SZ = DP_SZ[0]
+    SDP = 100. / signal.axes_manager.signal_axes[0].scale
+
+    offset2 = NX*NY + header['Data_offset_1']
+    # Based on inspected files, the DPs are stored at 16-bit boundary...
+    # Normally, you'd expect word alignment (32-bits) ¯\_(°_o)_/¯
+    offset2 += offset2 % 16
+
+    header = dict2sarray({
+        'NX': NX, 'NY': NY,
+        'DP_SZ': DP_SZ,
+        'SX': SX, 'SY': SY,
+        'SDP': SDP,
+        'Data_offset_2': offset2,
+        }, sarray=header)
+    return header, note
+
+
+def file_reader(filename, endianess='<', load_to_memory=False, mmap_mode='c',
+                **kwds):
+    metadata = {}
+    f = open(filename, 'rb')
+    header = np.fromfile(f, dtype=get_header_dtype_list(endianess), count=1)
+    header = sarray2dict(header)
+    note = str(f.read(header['Data_offset_1'] - f.tell()))
+    header['Note'] = note
+    NX, NY = header['NX'], header['NY']
+    DP_SZ = header['DP_SZ']
+    if header['SDP']:
+        SDP = 100. / header['SDP']
+    else:
+        SDP = Undefined
+    original_metadata = {'blockfile_header': header}
+
+    # A Virtual BF/DF is stored first
+#    offset1 = int(header['DATA_OFFSET_1'][0])
+#    f.seek(offset1)
+#    data_pre = np.array(f.read(offset2 - offset1), dtype=endianess+'u1'
+#        ).squeeze().reshape((NX, NY), order='C').T
+#    print len(data_pre)
+
+    # Then comes actual blockfile
+    offset2 = header['Data_offset_2']
+    if load_to_memory:
+        f.seek(offset2)
+        data = np.fromfile(f, dtype=endianess+'u1')
+        data = data.reshape((NY, NX, DP_SZ*DP_SZ + 6))
+    else:
+        data = np.memmap(f, mode=mmap_mode, offset=offset2,
+                         dtype=endianess+'u1', shape=(NY, NX, DP_SZ*DP_SZ + 6))
+
+    # Every frame is preceeded by a 6 byte sequence (AA 55, and then a 4 byte
+    # integer specifying frame number)
+    data = data[:, :, 6:]
+    data = data.reshape((NY, NX, DP_SZ, DP_SZ), order='C')
+
+    units = ['nm', 'cm', 'cm', 'nm']
+    names = ['x', 'dy', 'dx', 'y']
+    scales = [header['SX'], SDP, SDP, header['SY']]
+    metadata = {'General': {'original_filename': os.path.split(filename)[1]},
+                "Signal": {'signal_type': "",
+                           'record_by': 'image', },
+                }
+    # create the axis objects for each axis
+    dim = 4
+    axes = [
+        {
+            'size': data.shape[i],
+            'index_in_array': i,
+            'name': names[i + 3 - dim],
+            'scale': scales[i + 3 - dim],
+            'offset': 0.0,
+            'units': units[i + 3 - dim], }
+        for i in xrange(dim)]
+
+    dictionary = {'data': data,
+                  'axes': axes,
+                  'metadata': metadata,
+                  'original_metadata': original_metadata,
+                  'mapping': mapping, }
+
+    return [dictionary, ]
+
+
+def file_writer(filename, signal, **kwds):
```

Comment:
So far we haven't supported writing to proprietary formats because we can't guarantee that the proprietary software will be able to read the file.  However, I see that it can be useful in this and other cases. What about printing a warning so that the user is aware that the file is proprietary and we can't guarantee interoperability?